src/optimum/intel/neural_compressor/utils.py

```diff
@@ -97,3 +102,61 @@ def _get_quantizable_ops_recursively(
         else:
             self._get_quantizable_ops_recursively(child, op_name, quantizable_ops)
 
+
+def remove_inputs_from_graph(gm_original: GraphModule, inputs_to_remove: List[str]) -> GraphModule:
+    """
+    Remove specified inputs from a GraphModule.
+
+    Args:
+        gm_original (:obj:`GraphModule`):
+            Original GraphModule.
+        inputs_to_remove (:obj:`List[str]`):
+            List of string specifying the name of the inputs to remove from the GraphModule.
+    Returns:
+        gm (:obj:`GraphModule`):
+            GraphModule with the removed inputs.
+    """
+    from transformers.utils.fx_transformations import deepcopy_graph
+
+    try:
+        gm = deepcopy_graph(gm_original)
+        gm.eval()
+    except Exception as e:
+        gm = gm_original
+        logger.warning(f"Deepcopy failed: {repr(e)}, model is modified inplace.")
+
+    graph = gm.graph
+    output_node = None
+    for node in graph.nodes:
```

Comment:
I think you can simply retrieve the last node of the graph.nodes as it is topologically sorted (needs to be checked).